### DISCUSSION TITLE
Can't use name of AWS managed policy in Serverless::Function anymore

### DISCUSSION THREAD

**Discussion by kokobd** [jcouyang/dhall-aws-cloudformation#21] [General]
In https://github.com/jcouyang/dhall-aws-cloudformation/commit/5df088ea29dfc7b693d703ef04aa23d2304a2784#diff-a4d1e4b9b8d4a2350c8364f894fa27ce941261df804af6a0e61095c57dfec7e7R29-R35, the type of `Policies` is changed from `Optional (./../../Fn.dhall).CfnText` to 

```
Optional
          ( List
              < IAMPolicyDocument : (./IAMPolicyDocument.dhall).Type
              | SAMPolicyTemplate : (./SAMPolicyTemplate.dhall).Type
              >
          )
```

But AWS documentation says

> This property accepts a single value or list of values. Allowed values include:
> - [AWS SAM policy templates](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-policy-templates.html).
> - The ARN of an [AWS managed policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-vs-inline.html#aws-managed-policies) or [customer managed policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-vs-inline.html#customer-managed-policies).
> - **The name of an AWS managed policy from the following [list](https://github.com/aws/serverless-application-model/blob/develop/samtranslator/internal/data/aws_managed_policies.json).**
> - An [inline IAM policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-vs-inline.html#inline-policies) formatted in YAML as a map.

Previously we had simple json strings in the `Policies` list, like `Fn.renderText "AWSLambdaBasicExecutionRole"`. But now, given that each policy is a union, and the union doesn't have a constructor for strings, I think this usage has become impossible?

**Comment by endgame** [comment]
That will be because the code is faithfully generating from https://raw.githubusercontent.com/awslabs/goformation/master/generate/sam-2016-10-31.json . I'd file a bug against the goformation package and link it to this discussion, unless there's a true source further upstream.
